scripts/setup/mailchimp_connector.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import uuid
+from typing import Dict
+
+import requests
+import setup.constants as constants
+from setup import get_secret
+
+from fides.api.ops.api.v1 import urn_registry as urls
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def create_mailchimp_connector(
+    auth_header: Dict[str, str],
+    key: str = "mailchimp",
+):
+    if not key:
+        key = str(uuid.uuid4())
```

Comment:
Question about this: 

Am I correct that the only time that the conditional will evaluate to `true` is if the method is called as `create_mailchimp_connector(auth_header=header,key="")`, that is, because we have the default of `mailchimp`. 

If a function explicitly calls this with an empty string, I would expect it to fail instead of generating a uuid for it. Now if the value was omitted, it would then be fine to generate something random, but I would prefer to align the inputs with expected outputs.

Reply:
The `not key` check is Falsy, so `key=""` or `key=None` should both trigger it. If the DB object is created without a key, one will be automatically generated for it, so it's best we give it something sensible here. The default value was only meant to be a handy default.